### PR TITLE
Document email usage in the api

### LINKED ISSUE #6684: Document `email` on users
#6680 re-introduces the field "email" on User.  This field needs to be documented, along with "id" and "friendly_name".  For each field:

* What does it mean?
* When should it be used?  What should be done when the field is missing?

  For instance, we might say "To describe a user in a UI, use the first
  field out of friendly_name, email, id which is non-empty."
* Is it unique?
* Can it be modified?

There were numerous bugs with the previous incarnation of the undocumented
field named "email" on User.  Let's not go there again. :-)

### PR DESCRIPTION
**Pull request by itaiad200**
closes #6684

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]

File: api/swagger.yml [hunk 1]
```diff
@@ -672,15 +672,21 @@ components:
...
           description: Unix Epoch in seconds
         friendly_name:
           type: string
+          description: |
+            A shorter name for the user than the id. Unlike id it does not identify the user (it
+            might not be unique). Used in some places in the UI.
         email:
           type: string
+          description: |
+            The email address of the user. If API authentication is enabled, this field is mandatory and will be invited to login.
```
Comment (arielshaqed) [inline api/swagger.yml:688]: Is it unique? (I assume not, but it pays to be explicit)
Is it case sensitive?

**Review by arielshaqed — COMMENTED** [review]
This seems great and is an obvious improvement, a tonne of thanks for that! Not approving as I lack the definitive knowledge required to say it's correct.

**Review by guy-har — APPROVED** [review]
LGTM

File: api/swagger.yml [hunk 2]
```diff
@@ -672,15 +672,21 @@ components:
       properties:
         id:
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply api/swagger.yml:673]: @guy-har @Isan-Rivkin I need help answering this as the main consumers of this is us (Cloud, Enterprise)

File: api/swagger.yml [hunk 3]
```diff
@@ -672,15 +672,21 @@ components:
...
           type: string
-          description: a unique identifier for the user.
+          description: A unique identifier for the user.
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply api/swagger.yml:675]: Added that, thx

File: api/swagger.yml (hunk 1 above)
Comment (itaiad200) [reply api/swagger.yml:688]: @guy-har @Isan-Rivkin same here

File: api/swagger.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (guy-har) [reply api/swagger.yml:673]: Currently, the behavior in the UI isn't consistent
when listing users we display the first non-blank of email, id
when showing the current user in the navbar we display the first non-blank of friendly name, id

IMO we should align them both to the first non-blank of friendly name, email, and id but still provide a way to see the ID.
Having that said, is this something that should be documented or just a client decision?

File: api/swagger.yml (hunk 1 above)
Comment (guy-har) [reply api/swagger.yml:688]: lakeFS doesn't have such restrictions, in most cases we connect with an external auth provider which does cause the email to be unique

**Pull request merged**